src/analysis/tracker.rs

```diff
@@ -147,6 +149,60 @@ impl Tracker {
         Ok(())
     }
 
+    /// Check all of the extab relocations, and reject any invalid ones by checking against the decoded table data
+    /// of each table.
+    fn check_extab_relocations(&mut self, obj: &ObjInfo) -> Result<()> {
+        let mut to_reject = vec![];
+        let Some((section_index, section)) = obj.sections.by_name("extab")? else {
+            // No extab section found, return
+            return Ok(());
+        };
+        let relocs = &self.relocations;
+        for (_, symbol) in obj.symbols.for_section(section_index) {
+            let extab_name = &symbol.name;
+            let extab_start_addr: u32 = symbol.address as u32;
+            let extab_end_addr: u32 = extab_start_addr + symbol.size as u32;
+            let Ok(extab_data) = section.data_range(extab_start_addr, extab_end_addr) else {
+                log::warn!("Failed to get extab data for symbol {}", extab_name);
+                continue;
+            };
+            let data = match decode_extab(extab_data) {
+                Ok(decoded_data) => decoded_data,
+                Err(e) => {
+                    log::warn!(
+                        "Exception table decoding failed for symbol {}, reason: {}",
+                        extab_name,
+                        e.to_string()
+                    );
+                    return Ok(());
+                }
+            };
+            let mut decoded_reloc_addrs: Vec<u32> = vec![];
+            for reloc in data.relocations {
+                let reloc_addr = extab_start_addr + reloc.offset;
+                decoded_reloc_addrs.push(reloc_addr);
```

Comment:
A `BTreeSet` here would allow quicker lookup a `Vec`. The values are sorted on insertion.
```suggestion
            let mut decoded_reloc_addrs: BTreeSet<u32> = BTreeSet::new();
            for reloc in data.relocations {
                decoded_reloc_addrs.insert(extab_start_addr + reloc.offset);
```